onedal/linear_model/logistic_regression.py

```diff
@@ -0,0 +1,215 @@
+        m = module.model()
+
+        coefficients = self.coef_
+        dtype = get_dtype(coefficients)
+        if not isinstance(coefficients, np.ndarray):
+            coefficients = np.asarray(coefficients, dtype=dtype)
+
+        if coefficients.ndim == 2:
+            n_features_in = coefficients.shape[1]
+            n_targets_in = coefficients.shape[0]
+        else:
+            n_features_in = coefficients.size
+            n_targets_in = 1
+
+        intercept = self.intercept_
+        if isinstance(intercept, Number):
+            assert n_targets_in == 1
+        else:
+            if not isinstance(intercept, np.ndarray):
+                intercept = np.asarray(intercept, dtype=dtype)
+            assert n_targets_in == intercept.size
+
+        intercept = _check_array(
+            intercept,
+            dtype=[np.float64, np.float32],
+            force_all_finite=True,
+            ensure_2d=False,
+        )
+        coefficients = _check_array(
+            coefficients,
+            dtype=[np.float64, np.float32],
+            force_all_finite=True,
+            ensure_2d=False,
+        )
+
+        coefficients, intercept = make2d(coefficients), make2d(intercept)
+        coefficients = coefficients.T if n_targets_in == 1 else coefficients
+
+        assert coefficients.shape == (n_targets_in, n_features_in)
+        assert intercept.shape == (n_targets_in, 1)
+
+        desired_shape = (n_targets_in, n_features_in + 1)
+        packed_coefficients = np.zeros(desired_shape, dtype=dtype)
+
+        packed_coefficients[:, 1:] = coefficients
+        if self.fit_intercept:
+            packed_coefficients[:, 0][:, np.newaxis] = intercept
+
+        packed_coefficients = _convert_to_supported(policy, packed_coefficients)
+
+        m.packed_coefficients = to_table(packed_coefficients)
+
+        self._onedal_model = m
+
+        return m
+
+    def _predict(self, X, module, queue):
+        _check_is_fitted(self)
+
+        policy = self._get_policy(queue, X)
+
+        if isinstance(X, np.ndarray):
+            X_loc = np.asarray(X)
+        else:
+            X_loc = X
+
+        # Finiteness is checked in the sklearnex wrapper
+        X_loc = _check_array(
+            X_loc, dtype=[np.float64, np.float32], force_all_finite=False, ensure_2d=False
+        )
+        _check_n_features(self, X_loc, False)
+
+        if hasattr(self, "_onedal_model"):
+            model = self._onedal_model
+        else:
+            model = self._create_model(module, policy)
+
+        X_loc = make2d(X_loc)
+        X_loc = _convert_to_supported(policy, X_loc)
+        params = self._get_onedal_params(get_dtype(X_loc))
+
+        X_table = to_table(X_loc)
+        result = module.infer(policy, params, model, X_table)
+        y = from_table(result.responses)
+
+        if not isinstance(self.coef_, np.ndarray):
+            coefficients = np.asarray(self.coef_)
+        else:
+            coefficients = self.coef_
```

Comment:
From: https://numpy.org/doc/stable/reference/generated/numpy.asarray.html
> Array interpretation of a. No copy is performed if the input is already an ndarray with matching dtype and order. If a is a subclass of ndarray, a base class ndarray is returned.

Please remove this check. They just mess up the code.